src/server_manager/web_app/gcp_account.ts

```diff
@@ -84,10 +92,14 @@ export class GcpAccount implements gcp.Account {
   async listLocations(projectId: string): Promise<gcp.ZoneOption[]> {
     const listZonesResponse = await this.apiClient.listZones(projectId);
     const zones = listZonesResponse.items ?? [];
-    return zones.map(zoneInfo => ({
-      cloudLocation: new gcp.Zone(zoneInfo.name),
-      available: zoneInfo.status === 'UP'
-    }));
+    return zones.map(zoneInfo => {
+      const zone = new gcp.Zone(zoneInfo.name);
+      return {
+        cloudLocation: zone,
+        available: zoneInfo.status === 'UP',
+        lowerCost: FREE_TIER_REGIONS.has(zone.regionId),
```

Comment:
You need to update the `gcp.ZoneInfo` type.

You will want to update `filterOptions` to be a generic that takes <T extends CloudLocationOption> and returns T[] in order to not lose the typing.

In `GcpCreateServerApp.showRegionPicker()` you will want to convert to the UI input type.

Reply:
`zoneInfo` is a type defined by GCP.  Instead of returning the attribute here, I made it a check that GcpCreateServerApp can perform, and made `filterOptions` generic as suggested.